jenkins-ci/Jenkinsfile

```diff
@@ -0,0 +1,57 @@
+pipeline {
+    agent any
+    stages {
+        stage('all-in-one: clean environment') { 
+            steps { 
+                sh 'oc get all'
+                sh 'oc delete all,template,daemonset -l jaeger-infra'
+                sh 'oc get all'
+            }
+        }
+        stage('all-in-one: deploy Jaeger') { 
+            steps { 
+                sh 'oc get all'
+                sh 'oc process -f all-in-one/jaeger-all-in-one-template.yml | oc create -f -'
+                sh 'oc get all'
+            }
+        }
+        stage('all-in-one: verify deployment'){
+            steps {
+                sh 'oc get dc'
+                openshiftVerifyDeployment apiURL: '', authToken: '', depCfg: 'jaeger', namespace: '', replicaCount: '1', verbose: 'true', verifyReplicaCount: 'true', waitTime: '3', waitUnit: 'min'
+            }
+        }
+        stage('all-in-one: verify services'){
+            steps{
+                sh 'oc get services'
+                openshiftVerifyService apiURL: '', authToken: '', namespace: '', svcName: 'jaeger-query', verbose: 'true'
+            }
+        }
+        stage('all-in-one: functional test'){
+            steps{
+                withEnv(["JAVA_HOME=${ tool 'jdk8' }", "PATH+MAVEN=${tool 'maven-3.5.0'}/bin:${env.JAVA_HOME}/bin"]) {
+                    sh '''
+                        export JAEGER_QUERY_HOST="jaeger-query.jaeger-ci.svc"
+                        export JAEGER_AGENT_HOST="jaeger-agent.jaeger-ci.svc"
+                        export JAEGER_PORT_QUERY_HTTP=80
+                        export JAEGER_PORT_AGENT_ZIPKIN_THRIFT=5775
+                        export JAEGER_PORT_AGENT_COMPACT=6831
+                        export JAEGER_PORT_AGENT_BINARY=6832
+                        export JAEGER_PORT_ZIPKIN_COLLECTOR=14268
+                        rm ./jaeger-java-test -rf
+                        git clone https://github.com/Hawkular-QE/jaeger-java-test.git
```

Comment:
I might be missing something, but I think it would be awesome to run the Arquillian tests @pavolloffay authored instead, so that we can add our own tests in the future to that test suite and have it executed for every commit.